### PR TITLE
Adding Packages Api

### LINKED ISSUE #2360: Packages API
Hi, it looks like this library doesn't implement the [packages API](https://docs.github.com/en/rest/reference/packages)?

For example I wanted to delete some packages (which github doesn't support through `dotnet nuget delete`)

### PR DESCRIPTION
**Pull request by JonruAlveus**
fixes #2360

### CODE REVIEW

File: Octokit/IGitHubClient.cs
Comment (timrogers) [inline Octokit/IGitHubClient.cs:91]: *Packages

File: Octokit/Helpers/ApiUrls.cs
Comment (timrogers) [inline Octokit/Helpers/ApiUrls.cs]: *Packages

File: Octokit/GitHubClient.cs
Comment (timrogers) [inline Octokit/GitHubClient.cs:230]: *Packages

File: Octokit/Clients/IPackagesClient.cs
Comment (timrogers) [inline Octokit/Clients/IPackagesClient.cs]: Do you think `List` is the right name here? Should we use `ListForOrganization` or something like that, given that there are other list APIs?

File: Octokit.Reactive/Clients/ObservablePackagesClient.cs
Comment (timrogers) [inline Octokit.Reactive/Clients/ObservablePackagesClient.cs:72]: Should we be enforcing that the `packageName` is not null or empty, as we do for the `org`?

File: Octokit.Reactive/Clients/ObservablePackagesClient.cs
Comment (timrogers) [inline Octokit.Reactive/Clients/ObservablePackagesClient.cs:35]: C# newbie question: when the method is declared with argument `PackageType packageType`, is it implicitly possible to pass null?

**Review by timrogers — APPROVED** [review]
Thanks for taking the time to add this ❤️  I've reviewed commit-by-commit and it looks good to me from an API and basic implementation perspective, but I'm not a .NET expert, so let's wait for @nickfloyd to return from holiday before we merge (if that's okay!).

File: Octokit.Reactive/Clients/ObservablePackagesClient.cs
Comment (JonruAlveus) [reply Octokit.Reactive/Clients/ObservablePackagesClient.cs:72]: 👍

File: Octokit.Reactive/Clients/ObservablePackagesClient.cs
Comment (JonruAlveus) [reply Octokit.Reactive/Clients/ObservablePackagesClient.cs:35]: No probs! No, an enum is a value type (like an int) so it can only be null if it’s specified with a ?

I had to look that one up and I’ve been using c# for around a decade 😨

File: Octokit/IGitHubClient.cs
Comment (JonruAlveus) [reply Octokit/IGitHubClient.cs:91]: 👍

File: Octokit/Helpers/ApiUrls.cs
Comment (JonruAlveus) [reply Octokit/Helpers/ApiUrls.cs]: 👍

File: Octokit/GitHubClient.cs
Comment (JonruAlveus) [reply Octokit/GitHubClient.cs:230]: 👍

File: Octokit/Clients/IPackagesClient.cs
Comment (JonruAlveus) [reply Octokit/Clients/IPackagesClient.cs]: Yeah it wasn't the best name. I believe I changed it in later commits to "GetAllForOrg" and so on for the various combinations.

File: Octokit.Reactive/Clients/IObservablePackagesClient.cs
Comment (nickfloyd) [inline Octokit.Reactive/Clients/IObservablePackagesClient.cs:8]: So this is one of those weird API domains where package versions are a subset of packages, yet a package version is a package.

What you've done here feels like it works - I'm glad you approached the relationship this way to help reduce namespace confusion - I think leaving PackageVersions out of this interface would've made things a bit more clumsy.

```
github.Packages.PackageVersions.GetAllForUser
github.Packages.GetAllForOrg
```

feels more elegant than something like (though possible):

```
github.PackageVersions.GetAllForUser
github.Packages.GetAllForOrg
```

While, as you've implemented, it can lead to a bit more verbose API call, I think the expression feels more natural ❤️  solid work here!

**Review by nickfloyd — APPROVED** [review]
Yeah this all looks fantastic @JonruAlveus - other than the typos, this looks g2g! :heart:

File: Octokit.Reactive/Clients/IObservablePackagesClient.cs
Comment (JonruAlveus) [reply Octokit.Reactive/Clients/IObservablePackagesClient.cs:8]: Thanks! It is a little odd. I took my inspiration from the GitHub Api docs where it's all grouped together in one Packages section.

I've just pushed a final fix for the typos also.

**Comment by nickfloyd** [comment]
release_notes: Added Packages and Package versions APIs

**Pull request merged**
